app/src/parser.py

```diff
@@ -87,12 +87,13 @@ def extract(self, sheets):
                     # Process each row
                     row = df.loc[df["Wirtschaftsgliederung"] == branch]
                     if not row.empty:
-
                         # Extract value for each certain unit from the row
                         for unit in mapping_units:
                             if unit in row:
                                 value = row.iloc[0][unit]
-                                extracted[area][mapping_units[unit]][mapping_branches[branch]][year] = value
+                                extracted[area][mapping_units[unit]][
```

Comment:
This looks hard to read now, but I guess it's on me for not using variables instead of indexing everything on ther same line

Reply:
Hi @guadiromero I've rewritten the function using early "returns" (in that case continue for the for loops) to reduce identation and raise readability.

FYI the commit: https://github.com/technologiestiftung/innovationserhebung/pull/52/commits/d4f66e8585bdfd45e1b38498b111dd602a6feb32